### PR TITLE
Add pagination function to dropdown control

### PR DESCRIPTION
**Pull request by timtong1982**
#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [ ] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -178,6 +183,49 @@ export class Dropdown extends BaseComponent<IDropdownProps, IDropdownState> {
...
+
+  protected _setCalloutWidth(hostElement?: HTMLDivElement) {
+    if (this.props.pageSize && this.props.pageSize > 0) {
+      let totalItemCount = this._optionList.children.length;
+      if (this.props.pageSize < totalItemCount) {
+        var mainPanel = hostElement.querySelector('.ms-Callout-main') as HTMLDivElement;
+        if (mainPanel) {
+          if (mainPanel.offsetWidth > this._dropDown.clientWidth) {
+            let offset = mainPanel.offsetWidth - this._dropDown.clientWidth + 2;
+            this.setState({ scrollbarOffset: offset });
```
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: I don't think we really need scrollbarOffset, have you tried css `overflow` combinations to fix this ?

Comment (timtong1982) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: overflow: overlay works, but this only apply to webkit.

File: packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts [hunk 2]
```diff
@@ -106,6 +106,11 @@ export interface ICalloutProps extends React.Props<Callout | CalloutContent> {
   setInitialFocus?: boolean;
 
   /**
+   * Set the max height of Callout component.
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts]: What unit is this? Rows? Pixels?

**Comment by micahgodbolt** [comment]
We just pulled in a significant reworking of dropdown. Give me a shout if you have questions about the new approach when refactoring this.

File: packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts (hunk 2 above)
Comment (timtong1982) [inline packages/office-ui-fabric-react/src/components/Callout/Callout.Props.ts]: default is null, if nothing passed in here, the logic will use the existing logic:
 let contentMaxHeight: number = this.props.maxHeight || this._getMaxHeight();

**Comment by timtong1982** [comment]
David, please close this pull request since Dropdown change a lot now, I need re-create the code for this function.

**Pull request closed without merging**